src/core/mount.c

```diff
@@ -1056,6 +1054,8 @@ static void mount_enter_unmounting(Mount *m) {
                     MOUNT_UNMOUNTING_SIGKILL))
                 m->n_retry_umount = 0;
 
+        mount_unwatch_control_pid(m);
```

Comment:
Also this one. Please move the cleanups for `mount_enter_unmounting()` into another commit, or the commit for `mount_enter_mounting()`.